### PR TITLE
Add unit tests for cn.edu.jxnu.practice

### PR DESCRIPTION
**Pull request by ThomasPerkins1123**
Hi,

I've analysed your code base and noticed that `cn.edu.jxnu.practice` in the `cs-summary-reflection` module is not fully tested.

I've written some tests that cover this class with the help of [Diffblue Cover](https://www.diffblue.com/opensource).

Hopefully, these tests should help you detect any regressions caused by future code changes. If you would find it useful to have additional tests written for this repository, I would be more than happy to look at other particular classes that you consider important.

### CODE REVIEW

File: src/test/java/cn/edu/jxnu/practice/DifferentNumberOfBinaryBits_JavaTest.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package cn.edu.jxnu.practice;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class DifferentNumberOfBinaryBits_JavaTest {
```
Comment (petersear) [inline src/test/java/cn/edu/jxnu/practice/DifferentNumberOfBinaryBits_JavaTest.java]: Was there a reason you didn't combine the asserts into one test method?

File: src/test/java/cn/edu/jxnu/practice/RightShiftTest.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	public void testRightShift() {
+		int[] array = {};
+		RightShift.rightShift(new int[0], 1, 0);
+
+		Assert.assertArrayEquals(new int[0], array);
+
+		array = new int[]{1, 2, 5, 4, 5, 6, 7, 1, 9};
+		RightShift.rightShift(array, 8, 5);
+
+		Assert.assertArrayEquals(new int[]{4, 5, 6, 7, 1, 1, 2, 5, 9}, array);
```
Comment (petersear) [inline src/test/java/cn/edu/jxnu/practice/RightShiftTest.java]: line > 80 chars

File: src/test/java/cn/edu/jxnu/practice/RightShiftTest.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+	public void testRightShift2() {
+		int[] array = {};
+		RightShift.rightShift2(new int[0], 1, 0);
+
+		Assert.assertArrayEquals(new int[0], array);
+
+		array = new int[]{1, 2, 5, 4, 5, 6, 7, 1, 9};
+		RightShift.rightShift2(array, 8, 5);
+
+		Assert.assertArrayEquals(new int[]{4, 5, 6, 7, 1, 1, 2, 5, 9}, array);
```
Comment (petersear) [inline src/test/java/cn/edu/jxnu/practice/RightShiftTest.java]: line > 80 chars

File: src/test/java/cn/edu/jxnu/practice/StringDisplacementTest.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
+package cn.edu.jxnu.practice;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class StringDisplacementTest {
+
+	@Test
+	public void testStringDisplacement2() {
+		Assert.assertFalse(StringDisplacement.stringDisplacement2("2", "a/b/c"));
```
Comment (petersear) [inline src/test/java/cn/edu/jxnu/practice/StringDisplacementTest.java]: line > 80 chars (and lines following)

**Review by petersear — CHANGES_REQUESTED** [review]

**Review by paulkennethkent — APPROVED** [review]

File: src/test/java/cn/edu/jxnu/practice/DifferentNumberOfBinaryBits_JavaTest.java (hunk 1 above)
Comment (ThomasPerkins1123) [reply src/test/java/cn/edu/jxnu/practice/DifferentNumberOfBinaryBits_JavaTest.java]: No real reason I just didn't like only having one test in a class so I didn't merge them together but I have now

**Review by petersear — APPROVED** [review]

**Pull request merged**